test/05-device-state.spec.ts

```diff
@@ -283,14 +289,22 @@ describe('deviceState', () => {
 			.to.have.property('labels')
 			.that.has.property('io.balena.something')
 			.that.equals('bar');
+
+		// Restore renamed apps.json
+		await fsUtils.safeRename(appsJsonBackup(appsJson), appsJson);
```

Comment:
Is it possible to use mock-fs for these fs tests since we are trying to incrementally move all tests over to mock-fs anyway?

Reply:
I think I tried at some point for the v3 PR for this whole suite, but there were too many dependencies. It might be possible to do it just for preloaded files.

Reply:
That would be great if it's not too big a change!

Reply:
I remembered why I ended up not using mockfs. The `device-state` module has too many dependencies on config files, meaning if I cannot just mock `apps.json`, I have to mock `config.json`, `/etc/hostname`, etc.

Otherwise I start getting stuff like
```
 1) deviceState
       loads a target state from an apps.json file and saves it as target state, then returns it:
     Error: ENOENT, no such file or directory 'test/data/etc/hostname'
      at Binding.<anonymous> (node_modules/mock-fs/lib/binding.js:571:15)
```